### PR TITLE
Implemented and tested VR507-2

### PR DESCRIPTION
**Pull request by Mech0z**
## Description

Charge type is fee or subscription there may only be 1 price.

## References

* #194

### CODE REVIEW

**Review by bemwen — APPROVED** [review]

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Application/Validation/InputValidation/ValidationRules/ChargeFeeSubscriptionPriceCountRule.cs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using GreenEnergyHub.Charges.Domain.ChangeOfCharges.Transaction;
+
+namespace GreenEnergyHub.Charges.Application.Validation.InputValidation.ValidationRules
+{
+    public class ChargeFeeSubscriptionPriceCountRule : IValidationRule
```
Comment (BjarkeMeier) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Application/Validation/InputValidation/ValidationRules/ChargeFeeSubscriptionPriceCountRule.cs:19]: What is a `FeeSubscription`? ;-) I would consider splitting this rule into `FeeMustHaveSinglePriceRule` and `SubscriptionMustHaveSinglePriceRule`.

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/Application/Validation/InputValidation/ValidationRules/ChargeFeeSubscriptionPriceCountRuleTests.cs [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    public class ChargeFeeSubscriptionPriceCountRuleTests
+    {
+        [Theory]
+        [InlineAutoMoqData(ChargeType.Fee, 0, false)]
+        [InlineAutoMoqData(ChargeType.Fee, 1, true)]
+        [InlineAutoMoqData(ChargeType.Fee, 2, false)]
+        [InlineAutoMoqData(ChargeType.Subscription, 0, false)]
+        [InlineAutoMoqData(ChargeType.Subscription, 1, true)]
+        [InlineAutoMoqData(ChargeType.Subscription, 2, false)]
+        public void ChargePriceCountRule_WhenCalledWithPricePoints_Expects1PricePoint(
```
[the remark this answers is not in the document]
Comment (Mech0z) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/Application/Validation/InputValidation/ValidationRules/ChargeFeeSubscriptionPriceCountRuleTests.cs]: Fixed in 609d780

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Application/Validation/InputValidation/ValidationRules/ChargeFeeSubscriptionPriceCountRule.cs (hunk 1 above)
Comment (Mech0z) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Application/Validation/InputValidation/ValidationRules/ChargeFeeSubscriptionPriceCountRule.cs]: Split Fee and Subscription into 2 separate rules in

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Application/Validation/ValidationRuleIdentifier.cs [hunk 3]
```diff
@@ -40,5 +40,6 @@ public enum ValidationRuleIdentifier
         ChargeTypeTariffPriceCountRule, // VR507-1
         MaximumPriceRule, // VR509
         ChargePriceMaximumAndDecimalsDigitsRule, // VR457
+        FeeMustHaveSinglePriceRule, // VR507-2
```
Comment (lasrinnil) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Application/Validation/ValidationRuleIdentifier.cs:43]: You are missing the identifier for SubscriptionMustHaveSinglePriceRule  i believe

Comment (Mech0z) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Application/Validation/ValidationRuleIdentifier.cs:43]: Good catch, fixed in 5dafb76ba8d64b89deb324ab63da1c856336b89a

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/Application/Validation/InputValidation/ValidationRules/FeeMustHaveSinglePriceRuleTests.cs [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+            var sut = new FeeMustHaveSinglePriceRule(chargeCommand);
+
+            // Assert
+            sut.IsValid.Should().Be(expected);
+        }
+
+        [Theory]
+        [InlineAutoMoqData(ChargeType.Tariff)]
+        [InlineAutoMoqData(ChargeType.Unknown)]
+        public void IsValid_WhenFeeOrSubscription_ShouldParseValidation(
```
Comment (lasrinnil) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/Application/Validation/InputValidation/ValidationRules/FeeMustHaveSinglePriceRuleTests.cs:55]: Maybe I don't quite get this. But neither fee or subscription is parsed to the test?

Comment (Mech0z) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/Application/Validation/InputValidation/ValidationRules/FeeMustHaveSinglePriceRuleTests.cs]: Fixed in 584ba5e8fb8926370f5d95a85701de9a955d3ffe

**Review by lasrinnil — APPROVED** [review]

**Pull request merged**